src/rendergraph/common/rendergraph/types.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+namespace rendergraph {
+
+enum class PrimitiveType {
+    UInt,
+    Float,
+};
+
+enum class Type {
+    UInt,
+    Float,
+    Vector2D,
+    Vector3D,
+    Vector4D,
+    Matrix4x4
+};
+
+int sizeOf(Type type);
+int sizeOf(PrimitiveType primitiveType);
+
+template<typename T>
+Type typeOf();
+
+template<typename T>
+PrimitiveType primitiveTypeOf();
+
+template<typename T>
+int tupleSizeOf();
+
```

Comment:
these are all fairly cheap. Is there a disadvantage to inlining them? Then they could also be constexpr which might make sense (see my earlier request for more constexpr).